### DISCUSSION TITLE
SomfyPlus Unable to control RTS shades.

### DISCUSSION THREAD

**Discussion by markflorkowski** [HomeSpan/HomeSpan#620] [Q&A]
I have constructed the hardware as described in the [HomeSpan/SomfyPlus](https://github.com/HomeSpan/SomfyPlus#step-3-constructing-the-hardware) readme, and have confirmed with an RTL-SDR that it is broadcasting *a* signal on the correct frequency as expected when commands are issued, however I am not able to get it to control my somfy shades.  I also checked my (working) original remotes using the RTL-SDR, and they seem to be on slightly shifted frequecies. I attempted changing the frequency in `DEV_somfy.h` to match the frequencies I saw from the remotes and had no luck. 

Additionally, I used `rtl_fm` and [code from this repository](https://github.com/dimhoff/radio_stuff/tree/master) to attempt to decode the signals and confirm that they were working as expected. Using this, I was able to decode the signals from the official remotes, I was not able to decode the ones from the SomfyPlus device.

Has anyone else run into similar issues pairing to their Somfy shades? Any assistance would be greatly appreciated!

Thanks!

**Comment by HomeSpan** [comment]
What frequency is your remote using?

**Comment by markflorkowski** [comment]
Here's a capture of 5 of them, they seem to all fall within 20kHz of the 433.42Mhz frequency specified:

![image](https://github.com/HomeSpan/HomeSpan/assets/11494384/fda41731-723e-4baa-982f-e7d0b915e9fa)

The SomfyPlus is clearly centered on 433.42MHz, and seems to "bleed" into adjacent frequencies a lot more:

![image](https://github.com/HomeSpan/HomeSpan/assets/11494384/14127c70-13e5-4074-86cf-6307a3eff28c)

**Comment by HomeSpan** [comment]
I know the Somfy system is sensitive to frequencies, but I'm not sure if the spread you are seeing is causing the issue.  Are you using the hardware from Adafruit?

**Reply by markflorkowski** [reply]
Yeah, I ordered exactly the hardware recommended in the readme, with the addition of a [featherwing tripler](https://www.adafruit.com/product/3417) to connect them all together. 

![IMG_2180](https://github.com/HomeSpan/HomeSpan/assets/11494384/b09d4a44-a541-41f0-aed4-61dde9d129c8)

**Comment by markflorkowski** [comment]
I did notice that the signals from the OEM somfy remotes seem to be much shorter, though the SomfyPlus software may be auto-repeating the signal a few times (I didn't actually look too deeply at the code to see if this was the case)
